tests/ops/op_math/test_sprod.py

```diff
@@ -0,0 +1,635 @@
+    qml.WireCut,
+)
+
+non_param_ops = (
+    (qml.Identity, gd.I),
+    (qml.Hadamard, gd.H),
+    (qml.PauliX, gd.X),
+    (qml.PauliY, gd.Y),
+    (qml.PauliZ, gd.Z),
+    (qml.S, gd.S),
+    (qml.T, gd.T),
+    (qml.SX, gd.SX),
+    (qml.CNOT, gd.CNOT),
+    (qml.CZ, gd.CZ),
+    (qml.CY, gd.CY),
+    (qml.SWAP, gd.SWAP),
+    (qml.ISWAP, gd.ISWAP),
+    (qml.SISWAP, gd.SISWAP),
+    (qml.CSWAP, gd.CSWAP),
+    (qml.Toffoli, gd.Toffoli),
+)
+
+param_ops = (
+    (qml.RX, gd.Rotx),
+    (qml.RY, gd.Roty),
+    (qml.RZ, gd.Rotz),
+    (qml.PhaseShift, gd.Rphi),
+    (qml.Rot, gd.Rot3),
+    (qml.U1, gd.U1),
+    (qml.U2, gd.U2),
+    (qml.U3, gd.U3),
+    (qml.CRX, gd.CRotx),
+    (qml.CRY, gd.CRoty),
+    (qml.CRZ, gd.CRotz),
+    (qml.CRot, gd.CRot3),
+    (qml.IsingXX, gd.IsingXX),
+    (qml.IsingYY, gd.IsingYY),
+    (qml.IsingZZ, gd.IsingZZ),
+)
+
+ops = (
+    (1.0, qml.PauliX(wires=0)),
+    (0.0, qml.PauliZ(wires=0)),
+    (1j, qml.Hadamard(wires=0)),
+    (1.23, qml.CNOT(wires=[0, 1])),
+    (4.56, qml.RX(1.23, wires=1)),
+    (1.0 + 2.0j, qml.Identity(wires=0)),
+    (10, qml.IsingXX(4.56, wires=[2, 3])),
+    (0j, qml.Toffoli(wires=[1, 2, 3])),
+    (42, qml.Rot(0.34, 1.0, 0, wires=0)),
+)
+
+ops_rep = (
+    "1.0*(PauliX(wires=[0]))",
+    "0.0*(PauliZ(wires=[0]))",
+    "1j*(Hadamard(wires=[0]))",
+    "1.23*(CNOT(wires=[0, 1]))",
+    "4.56*(RX(1.23, wires=[1]))",
+    "(1+2j)*(Identity(wires=[0]))",
+    "10*(IsingXX(4.56, wires=[2, 3]))",
+    "0j*(Toffoli(wires=[1, 2, 3]))",
+    "42*(Rot(0.34, 1.0, 0, wires=[0]))",
+)
+
+
+def get_qft_mat(num_wires):
+    """Helper function to generate the matrix of a qft protocol."""
+    omega = math.exp(np.pi * 1.0j / 2 ** (num_wires - 1))
+    mat = math.zeros((2**num_wires, 2**num_wires), dtype="complex128")
+
+    for m in range(2**num_wires):
+        for n in range(2**num_wires):
+            mat[m, n] = omega ** (m * n)
+
+    return 1 / math.sqrt(2**num_wires) * mat
+
+
+class TestInitialization:
+    """Test initialization of ther SProd Class."""
+
+    @pytest.mark.parametrize("test_id", ("foo", "bar"))
+    def test_init_sprod_op(self, test_id):
+        sprod_op = s_prod(3.14, qml.RX(0.23, wires="a"), do_queue=True, id=test_id)
+
+        assert (
+            sprod_op.scalar == 3.14
+        )  # no need to test if op.base == RX since this is covered in SymbolicOp tests
+        assert sprod_op.wires == Wires(("a",))
+        assert sprod_op.num_wires == 1
+        assert sprod_op.name == "SProd"
+        assert sprod_op.id == test_id
+
+        assert sprod_op.data == [[3.14], [0.23]]
+        assert sprod_op.parameters == [[3.14], [0.23]]
+        assert sprod_op.num_params == 2
+
+    def test_queue_idx(self):
+        sprod_op = s_prod(3.14, qml.Identity(wires=1))
+        assert sprod_op.queue_idx is None
+
+    def test_parameters(self):
+        sprod_op = s_prod(9.87, qml.Rot(1.23, 4.0, 5.67, wires=1))
+        assert sprod_op.parameters == [[9.87], [1.23, 4.0, 5.67]]
```

Comment:
Given that the `parameters` property is defined in the `Operator` class, is it strictly necessary to test it here?

Reply:
Always good to over-test things, especially if they're cheap to test. Testing is not as much about "strictly necessary" and more about "what might be a hidden problem at point long into the future?"